### PR TITLE
fix oracle linux 9 repository definitions

### PR DESCRIPTION
**Pull request by mcalmer**
## What does this PR change?

- repourl was wrongly pointing to a GPG key and the GPG key field was empty
- fixed also the distchannelmap setting

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- No tests: already covered

- [x] **DONE**

## Links

Fixes https://github.com/uyuni-project/uyuni/issues/6266
Fixes https://github.com/SUSE/spacewalk/issues/19799

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Review by juliogonzalez — CHANGES_REQUESTED** [review]
Change LGTM; but we need a schema migration to fix this for the people that got it broken.

File: schema/spacewalk/upgrade/susemanager-schema-4.4.3-to-susemanager-schema-4.4.4/000-uyuni-fix-ol9-repos.sql [hunk 1]
```diff
@@ -0,0 +1,36 @@
+INSERT INTO rhnContentSource (id, org_id, type_id, source_url, label, metadata_signed)
+  SELECT sequence_nextval('rhn_chan_content_src_id_seq'), CS.org_id, CS.type_id,
+         'https://yum.oracle.com/repo/OracleLinux/OL9/baseos/latest/x86_64/', 'Extern - Oracle Linux 9 (x86_64)', 'N'
```
Comment (juliogonzalez) [inline schema/spacewalk/upgrade/susemanager-schema-4.4.3-to-susemanager-schema-4.4.4/000-uyuni-fix-ol9-repos.sql:3]:
```suggestion
         'https://yum.oracle.com/repo/OracleLinux/OL9/baseos/latest/x86_64/', 'External - Oracle Linux 9 (x86_64)', 'N'
```

image

File: schema/spacewalk/upgrade/susemanager-schema-4.4.3-to-susemanager-schema-4.4.4/000-uyuni-fix-ol9-repos.sql [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    FROM rhnContentSource CS
+    JOIN rhnChannelContentSource CCS ON CS.id = CCS.source_id
+    JOIN rhnChannel C ON CCS.channel_id = C.id
+   WHERE CS.source_url = 'file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle'
+     AND C.label = 'oraclelinux9-x86_64';
+
+INSERT INTO rhnContentSource (id, org_id, type_id, source_url, label, metadata_signed)
+  SELECT sequence_nextval('rhn_chan_content_src_id_seq'), CS.org_id, CS.type_id,
+         'https://yum.oracle.com/repo/OracleLinux/OL9/baseos/latest/aarch64/', 'Extern - Oracle Linux 9 (aarch64)', 'N'
```
Comment (juliogonzalez) [inline schema/spacewalk/upgrade/susemanager-schema-4.4.3-to-susemanager-schema-4.4.4/000-uyuni-fix-ol9-repos.sql:12]:
```suggestion
         'https://yum.oracle.com/repo/OracleLinux/OL9/baseos/latest/aarch64/', 'External - Oracle Linux 9 (aarch64)', 'N'
```

**Review by juliogonzalez — CHANGES_REQUESTED** [review]

File: schema/spacewalk/upgrade/susemanager-schema-4.4.3-to-susemanager-schema-4.4.4/000-uyuni-fix-ol9-repos.sql (hunk 1 above)
Comment (mcalmer) [reply schema/spacewalk/upgrade/susemanager-schema-4.4.3-to-susemanager-schema-4.4.4/000-uyuni-fix-ol9-repos.sql:3]: No, this does not work. We need a different name. There is a unique index on it

File: schema/spacewalk/upgrade/susemanager-schema-4.4.3-to-susemanager-schema-4.4.4/000-uyuni-fix-ol9-repos.sql (hunk 2 above)
Comment (mcalmer) [reply schema/spacewalk/upgrade/susemanager-schema-4.4.3-to-susemanager-schema-4.4.4/000-uyuni-fix-ol9-repos.sql:12]: fixed with a final update statement

**Review by juliogonzalez — APPROVED** [review]
LGTM

**Pull request merged**